js/components/portal-dashboard/all-responses-popup/popup-header.tsx

```diff
@@ -0,0 +1,85 @@
+            responsesToggleClass: any,
+            closeIconClass: any;
+        if (!this.state.inFeedbackMode) {
+                feedbackToggleClass =css.feedbackToggle+ " " + css.toggleOff;
+                popupHeaderClass = css.popupHeader;
+                responsesToggleClass = css.responseToggle;
+                closeIconClass = css.closeIcon;
+        } else {
+            feedbackToggleClass = css.feedbackToggle;
+            popupHeaderClass = css.popupHeader + " " + css.inFeedbackMode;
+            responsesToggleClass = css.responseToggle + " " + css.toggleOff;
+            closeIconClass = css.closeIcon + " " + css.inFeedbackMode;
+        }
+        return (
+            <div className={popupHeaderClass} data-cy="popup-header">
+                <div className={css.headerLeft}>
+                    <svg className={`${css.assignmentIcon} ${css.icon}`}>
+                        <use xlinkHref={"#icon-assignment"} />
+                    </svg>
+                    <div className={css.title} data-cy="popup-header-title">{activityName}</div>
+                </div>
+                <div className={`${css.toggleResponsesFeedback}`}>
+                    <div className={responsesToggleClass} data-cy="all-students-responses-toggle" onClick={this.handleResponsesToggleClick}>
+                        <svg className={`${css.icon} ${css.toggleIcon} ${css.responseIcon}`}>
+                            <use xlinkHref={"#icon-group"} />
+                        </svg>
+                        <div className={`${css.toggleTitle} ${css.responseTitle} `}>All Student Responses</div>
+                    </div>
+                    <div className={feedbackToggleClass} data-cy="feedback-toggle" onClick={this.handleFeedbackToggleClick}>
+                        <div className={`${css.toggleTitle} ${css.feedbackTitle}`}>Feedback to Students</div>
+                        <svg className={`${css.icon} ${css.toggleIcon} ${css.feedbackIcon}`}>
+                            <use xlinkHref={"#icon-feedback-button"} />
+                        </svg>
+                    </div>
+                </div>
+                <div className={css.headerRight}>
+                    <div className={closeIconClass} data-cy="close-popup-button" onClick={this.handleCloseAllResponsesButtonClick}>
+                        <svg className={css.closeIconSVG}>
+                            <use xlinkHref={"#icon-close"} />
+                        </svg>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    private handleFeedbackToggleClick=()=>{
+        if(!this.state.inFeedbackMode){
+            this.setState({
+                inFeedbackMode: !this.state.inFeedbackMode
+              });
+        }
+    }
+    private handleResponsesToggleClick=()=>{
+        if(this.state.inFeedbackMode){
+            this.setState({
+                inFeedbackMode: !this.state.inFeedbackMode
+              });
+        }
+    }
```

Comment:
handleFeedbackToggleClick and handleResponsesToggleClick do the same thing.  Let's use a single function.

Reply:
Removed the function and set the state in an inline function